google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2000-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.facet;
+
+import com.google.cloud.tools.intellij.stats.UsageTrackerProvider;
+import com.google.cloud.tools.intellij.util.GctTracking;
+import com.google.common.collect.Sets;
+
+import com.intellij.facet.Facet;
+import com.intellij.facet.ui.FacetEditorContext;
+import com.intellij.facet.ui.FacetEditorTab;
+import com.intellij.openapi.roots.DependencyScope;
+import com.intellij.openapi.roots.ExportableOrderEntry;
+import com.intellij.openapi.roots.JavaProjectModelModificationService;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.openapi.roots.OrderEntry;
+import com.intellij.openapi.roots.libraries.Library;
+import com.intellij.openapi.roots.libraries.LibraryTable.Listener;
+import com.intellij.openapi.roots.libraries.LibraryTablesRegistrar;
+import com.intellij.packaging.artifacts.Artifact;
+
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+import java.util.Set;
+
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+
+/**
+ * @author nik
+ */
+public class AppEngineStandardFacetEditor extends FacetEditorTab {
+
+  private final AppEngineFacetConfiguration facetConfiguration;
+  private final FacetEditorContext context;
+  private JPanel myMainPanel;
+  private AppEngineStandardLibraryPanel appEngineStandardLibraryPanel;
+  private Listener libraryListener;
+
+
+  public AppEngineStandardFacetEditor(AppEngineFacetConfiguration facetConfiguration,
+      FacetEditorContext context) {
+    this.facetConfiguration = facetConfiguration;
+    this.context = context;
+    libraryListener = new LibraryModificationListener();
+
+    LibraryTablesRegistrar.getInstance()
+        .getLibraryTable(context.getProject()).addListener(libraryListener);
+  }
+
+  @Nls
+  public String getDisplayName() {
+    return "Google App Engine";
```

Comment:
Ok. I think I will also rename this to Google App Engine Standard since this facet only applies to that env.